### DISCUSSION TITLE
404 NOT FOUND

### DISCUSSION THREAD

**Discussion by aclaracastro** [vercel/community#6980] [Help] [answered]
### Summary

when I go to deploy it goes straight to error 404 and I already checked if the url was correct and other things but, idk what I can do to resolve it

### Steps to Reproduce

![404](https://github.com/vercel/community/assets/103144032/bf66775c-022b-4db4-9bc8-fd48b3e11354)

**Comment by versecafe** [comment]
@aclaracastro Can you share your repo or give some more context like build logs?

**Reply by aclaracastro** [reply]
sure!

https://github.com/vercel/community/assets/103144032/8fed649b-f43b-4cc3-910d-4b12d942e140

**Reply by aclaracastro** [reply]
all the times i try to deploy another project give the same page :(

**Reply by versecafe** [reply]
This is expected behaviour you're deploying without a set supported framework so unless you manually configure a `vercel.json` which declares how the build outputs should be handled then your end result is a 404, if you need documentation on doing custom deploys to Vercel such as raw react + vite or something like Flutter Web I can find the relevant configuration you'll need.

**Reply by studentiyot** [reply]
Hi, I'm having the same problem. I did discover in the documentation this is a "common error," but no direction to follow. 

I'm a baby developer with minor issues in GitHub Pages rendering. The incidental "bug" in my Pages isn't worth a big time commitment rn, but I wanted to see my beginner project in a different environment. Someone suggested Vercel is "newbie friendly," so I jumped in. If this isn't the case, I'll wait till  I'm more advanced in frameworks or Vercel comes up in a tutorial.

I can understand the gist of this reply—setting up a manual `vercel.json`— but I don't see a path to help noobs. Please direct me to basic install guidance for new users.. or I'll let it go. TY.

---

_EDIT: Oh, I want to add that the confetti at "launch" on the Vercel Dashboard just killed me when I could plainly see in the preview that all was not well. Lol._

**Comment by andromidasj** [comment]
Can confirm that I'm getting this issue as well. My project is deployed with the NextJS preset, and I have a custom 404 page located in `/src/app/not-found.tsx`. It works perfectly fine locally, both with `next dev` and with `next build && next start`, but in the live Vercel deployments (both the `project-name.vercel.app` and the `my-custom-domain.com`) I get the same error.

| Local | Production |
|--------|--------|
| ![CleanShot 2024-05-26 at 14 34 34@2x](https://github.com/vercel/community/assets/66661368/8437a7a8-01c1-4079-b077-caba0fc5f8a8) | ![CleanShot 2024-05-26 at 14 56 17@2x](https://github.com/vercel/community/assets/66661368/bba54823-936a-4100-acc8-986c10dca703) |

**Comment by amyegan** [comment, marked as answer]
Hi everyone! A [404 NOT_FOUND](https://vercel.com/docs/errors/NOT_FOUND#not_found) means the requested resource could not be found. There are too many possible causes for me to be able to offer a single solution for everyone, but here are some things you can check to narrow down the cause in your case:

- Check the deployment URL for typos or incorrect paths.
- Make sure the [deployment exists](https://vercel.com/docs/projects/project-dashboard#deployments) and has not been deleted.
- If the deployment exists, review the [deployment logs](https://vercel.com/docs/deployments/logs) for issues that might have caused the deployment to be unavailable.
- From the deployment details page > Source tab, toggle to Output, and verify that an index is available. This is sometimes missing. 
  - I've seen it happen with some Express.js setups, for example, and I recommend the [Express on Vercel](https://vercel.com/guides/using-express-with-vercel) guide in those cases. 
  - Projects not using a framework sometimes run into this as well. In that case, just make sure your site has an `index.html` file at the root of the project for the homepage.
- Check the [Runtime Logs](https://vercel.com/docs/observability/runtime-logs#runtime-logs) for errors preventing pages from being available.
- If you're on a team with multiple people, make sure you have [permissions](https://vercel.com/docs/accounts/team-members-and-roles) to access the deployment.

https://vercel.com/docs/errors/NOT_FOUND#not_found

**Reply by studentiyot** [reply]
Thank you very much for the reply.

_Oh my goodness.. "verify that an index is available." `<face plant>`  Lol._ 

The reason I came to Vercel (specifically from GitHub) is because of a minor issue with GH Pages converting markdown to HTML in the README file. I've been so focused on the README.md it didn't even occur to me to have an index (either `.md` or HTML). Yes, this fixed my issue. I'm still way over my skis here, but at least I have a functioning deployment. **Ok, throw me some confetti!!** 🎉 


For what it's worth—I'm coming in as a beginner, so excuse my lack of depth in providing context for the error. In my case, I initiated a new Vercel account, signing in from a GitHub account. I was asked to authenticate and plug in my credentials, so I expected the integration to be straightforward. There was nothing to type or have misnamed. The domain/project shows up in the dashboard—just with a sad 404... ~I think some kind of config is clearly missing, but I have no idea where to start.~
`<never mind>`
